### PR TITLE
Refactor app-ecs-services into a module and per-environment projects

### PR DESCRIPTION
**Pull request by issyl0**
# Why I am making this change

Refactoring the Terraform projects will enable us to run terraform commands directly, without wrapper scripts and source environment_{environment}.sh to set .tfvars.

# What approach I took

Much the same as #218 etc.

Paired with @matthewcullum-gds.

### CODE REVIEW

**Comment by philandstuff** [comment]
>  (ie the ones which people deploy using aws-vault and the ones people deploy using deploy.sh)

I fix this problem by never using `deploy.sh` 😄

File: terraform/projects/app-ecs-services-dev/main.tf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+variable "aws_region" {
+  type        = "string"
+  description = "AWS region"
+  default     = "eu-west-1"
+}
+
+variable "dev_environment" {
+  type        = "string"
+  description = "Boolean flag for development environments"
+  default     = "false"
```
[the remark this answers is not in the document]
Comment (philandstuff) [reply terraform/projects/app-ecs-services-dev/main.tf]: is there a reason to have `variable`s in projects at all, rather than just `local`s?

**Comment by issyl0** [comment]
Should I delete `deploy.sh`?

File: terraform/projects/app-ecs-services-dev/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (issyl0) [reply terraform/projects/app-ecs-services-dev/main.tf]: Lift and shift. 😉

**Comment by issyl0** [comment]
@idavidmcdonald No, we don't differentiate any more between enclave and non-enclave. Bye bye Makefile and hello `terraform ...`.

**Comment by idavidmcdonald** [comment]
There are references to `deploy.sh` in https://github.com/alphagov/prometheus-aws-configuration-beta/tree/terraform_module_app-ecs-services/terraform/modules/enclave still I think

**Comment by philandstuff** [comment]
I think we can also delete the `stacks/` directory and the references to it in `.gitignore` 🚮

File: terraform/modules/enclave/README.md [hunk 2]
```diff
@@ -27,15 +26,12 @@ Follow these steps to run infrastructure tests:
...
 
-    ./deploy.sh -e <environment> -p <aws vault profile> -a <terraform method> -s <state> -t <target>
-
-`<environment>` can only be one of: `paas-staging` or
-`paas-production` (unless it's your EC2 dev stack).  `<state>` must be
-`prometheus`.
-
-`<target>` is optional an example target module would be `module.prometheus.aws_instance.prometheus[0]`. This would deploy to the first Prometheus instance.
+```shell
+cd ../projects/enclave/paas-staging
```
Comment (philandstuff) [inline terraform/modules/enclave/README.md]:
```suggestion
cd projects/enclave/paas-staging/prometheus
```

**Pull request merged**